### PR TITLE
feat(unlock-app) - currency for lock frontend

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

Allow the Lock Manager to change the currency used for the credit card flow. 

needs https://github.com/unlock-protocol/unlock/pull/12245 to be merged

**USD currency**

Screenshot 2023-06-27 at 18 06 47

**EUR currency**

Screenshot 2023-06-27 at 18 07 14

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCurrency.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      value: currency,
+    }
+  })
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className="grid items-center gap-2">
+      <div className="flex flex-col gap-2">
+        <span className="text-base font-bold text-gray-700">Currency</span>
+        <Select
+          description="The selected currency will be used for credit card payments. (the currency is automatically updated when the selection is changed)"
```
Comment (kalidiagne) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardCurrency.tsx:61]: @julien51 any improvement that we can make here?

**Comment by julien51** [comment]
Please move the choice of currency to be next to the field with the fixed price for clarity.

**Comment by kalidiagne** [comment]
> Please move the choice of currency to be next to the field with the fixed price for clarity.

Update, can can you please check?

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx [hunk 2]
```diff
@@ -141,25 +172,43 @@ export default function CreditCardCustomPrice({
...
-              })}
-            />
+            <div className="grid gap-2 md:grid-cols-2">
+              <div className="flex flex-col gap-1">
+                <span>Price</span>
+                <Input
+                  type="number"
+                  step="any"
+                  disabled={disabled}
+                  description="Set a fixed price in USD that is charged for card payments."
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx:182]:
```suggestion
                  description="Set a fixed price in fiat currency that is charged for card payments."
```

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx [hunk 3]
```diff
@@ -141,25 +172,43 @@ export default function CreditCardCustomPrice({
...
+                  error={errors?.creditCardPrice?.message}
+                  {...register('creditCardPrice', {
+                    required: {
+                      value: toggleActive,
+                      message: 'This field is required.',
+                    },
+                    min: {
+                      value: CREDIT_CARD_MIN_USD_PRICE,
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx:190]: I wonder what happens if the user sets 0.1 EUR for example?

**Review by julien51 — COMMENTED** [review]
The approach looks good but I think there is an issue when the price is too low in Euros...

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx (hunk 3 above)
Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx:190]: in this case, the user puts `0.1` will show the error, we only consider price > `0.5` here, 

but let me refactor and add this: 

https://stripe.com/docs/currencies

**Comment by kalidiagne** [comment]
> The approach looks good but I think there is an issue when the price is too low in Euros...

Added mapping for min prices, in out case `USD` & `EUR` pricing have the min price on stripe side, but the mapping will help with new currencies

**Comment by kalidiagne** [comment]
To close the loop also the currency in the checkout 

**EURO**

Screenshot 2023-06-27 at 21 21 53

**USD** 

Screenshot 2023-06-27 at 21 22 42

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx (hunk 3 above)
Comment (julien51) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx]: Yes that's exactly what we need!

**Review by julien51 — APPROVED** [review]

**Pull request merged**